src/policy/space.rs

```diff
@@ -222,7 +214,9 @@ impl SFTMap {
                 new
             );
         }
-        self_mut.sft[chunk] = sft;
+        // We have an assumption that a Space is static that we never drop a space. Under this assumption, the following cast is correct.
```

Comment:
How about changing this argument?:

```
sft: &(dyn SFT + Sync + 'static)
```

to

```
sft: &'static (dyn SFT + Sync + 'static)
```

Reply:
This needs changes to some other functions' arguments to compile, but it doesn't need the unsafe code here

Reply:
I tried this. If we keep propagating the `static` requirement, eventually we need the `MMTK`->`Plan`->`Space` all to be static. But `MMTK` is instantiated in the binding, and we cannot safely know `MMTK` is static. We would also need unsafe somewhere. In that case, I think it might be better to do the unsafe code here, just in one place. Let me know what you think.

Reply:
I tried doing this before writing the comment and I could do it without any unsafe.
I don't have the CL right now, but I will try to do it again and see if I did anything wrong, or it is possible without unsafe

Reply:
Sure. It will be good if we can remove this unsafe code.

Reply:
I pushed my proposed changes, take a look and revert changes if you think anything is wrong

Reply:
That looks great. I will work on your changes. I will ask you for another review/approval when I am done.